packages/apollo-engine-reporting/src/plugin.ts

```diff
@@ -223,11 +271,22 @@ export const plugin = <TContext>(
             treeBuilder.trace.clientName = clientName || '';
           }
         },
+        async didResolveOperation(requestContext) {
+          await shouldTraceOperation(requestContext);
 
+          if (metrics.captureTraces === false) {
+            // End early if we aren't going to send the trace so we continue to
+            // run the tree builder.
+            didEnd(requestContext);
+          }
+        },
         executionDidStart() {
+          // If we stopped tracing early, return to avoid creating the listener.
+          if (endDone) return;
+
           return {
             willResolveField({ info }) {
-              return treeBuilder.willResolveField(info);
+              treeBuilder.willResolveField(info);
```

Comment:
why is the `return` gone from here?  I think that means we end up losing the end callback it returns? I don't think there's implicit return in TypeScript in methods that are defined with `{blocks}` but maybe I'm wrong.

Reply:
Yep you are right. Fixed the bug.